        -:    0:Source:main.cpp
        -:    0:Graph:main.gcno
        -:    0:Data:main.gcda
        -:    0:Runs:1
        -:    0:Programs:1
        -:    1:#include <iostream>
        -:    2:#include "fly.h"
        -:    3:using namespace std;
        -:    4:
        -:    5:class Foo {
        -:    6:public:
function _ZN3Foo3fooEv called 1 returned 100% blocks executed 100%
        1:    7:  void foo() {
        1:    8:    cout << __FILE__ << endl;
call    0 returned 100%
call    1 returned 100%
        1:    9:    cout << __LINE__ << endl;
call    0 returned 100%
call    1 returned 100%
        1:   10:    cout << __FUNCTION__ << endl;
call    0 returned 100%
call    1 returned 100%
        1:   11:  }
        -:   12:
function _ZN3Foo3fooEi called 1 returned 100% blocks executed 100%
        1:   13:  void foo(int x) {
        1:   14:    cout << __FILE__ << endl;
call    0 returned 100%
call    1 returned 100%
        1:   15:    cout << __LINE__ << endl;
call    0 returned 100%
call    1 returned 100%
        1:   16:    cout << __FUNCTION__ << endl;
call    0 returned 100%
call    1 returned 100%
        1:   17:  }
        -:   18:};
        -:   19:
        -:   20:class Foo2 {
        -:   21:public:
function _ZN4Foo23fooEv called 1 returned 100% blocks executed 100%
        1:   22:  void foo() {
        1:   23:    cout << __FILE__ << endl;
call    0 returned 100%
call    1 returned 100%
        1:   24:    cout << __LINE__ << endl;
call    0 returned 100%
call    1 returned 100%
        1:   25:    cout << __FUNCTION__ << endl;
call    0 returned 100%
call    1 returned 100%
        1:   26:  }
        -:   27:};
        -:   28:
function main called 1 returned 100% blocks executed 100%
        1:   29:int main() {
        -:   30:  Foo foo;
        1:   31:  foo.foo();
call    0 returned 100%
branch  1 taken 100% (fallthrough)
branch  2 taken 0% (throw)
        1:   32:  foo.foo(1);
call    0 returned 100%
branch  1 taken 100% (fallthrough)
branch  2 taken 0% (throw)
        -:   33:  Foo2 foo2;
        1:   34:  foo2.foo();
call    0 returned 100%
branch  1 taken 100% (fallthrough)
branch  2 taken 0% (throw)
        1:   35:  Fly fly;
call    0 returned 100%
branch  1 taken 100% (fallthrough)
branch  2 taken 0% (throw)
        1:   36:  return 0;
function _GLOBAL__sub_I_main called 1 returned 100% blocks executed 100%
function _Z41__static_initialization_and_destruction_0ii called 1 returned 100% blocks executed 100%
        3:   37:}
call    0 returned 100%
branch  1 taken 100% (fallthrough)
branch  2 taken 0%
branch  3 taken 100% (fallthrough)
branch  4 taken 0%
        -:   38:
